rayon-core/src/registry.rs

```diff
@@ -520,6 +536,19 @@ unsafe fn main_loop(worker: Worker<JobRef>, registry: Arc<Registry>, index: usiz
 
     // Normal termination, do not abort.
     mem::forget(abort_guard);
+
+    // Inform a user callback that we exited a thread.
```

Comment:
e.g. if a job placed in the thread pool panicked?  I think the answer is "no".  We could certainly reach this code if some non-thread pool threads panicked, though.